apps/mobile/src/components/Fiat/Fiat.tsx

```diff
@@ -23,25 +27,25 @@ export const Fiat = ({ value, currency, maxLength, precise }: FiatProps) => {
     return match ? match.slice(1) : ['', '', preciseFiat, '', '']
   }, [preciseFiat])
 
-  if (fiat == null) {
-    return <H1 fontWeight="600">--</H1>
-  }
-
   return (
     <View flexDirection="row" alignItems="center" testID={'fiat-balance-display'}>
       {precise ? (
-        <XStack>
-          <H2 fontWeight={'600'} alignSelf={'flex-end'} marginBottom={'$2'}>
-            {symbol}
-          </H2>
-          <H1 fontWeight="600">{whole}</H1>
-          {decimals && (
-            <H1 fontWeight={600} color="$textSecondaryDark">
-              {decimals}
-            </H1>
-          )}
-          <H1 fontWeight={600}>{endCurrency}</H1>
-        </XStack>
+        <Skeleton.Group show={!fiat}>
+          <Skeleton colorMode={colorScheme === 'dark' ? 'dark' : 'light'} width={220}>
```

Comment:
I don't understand when this skeleton kicks in? 

fiat always end up being a string, so the Skeleton will never kick in? I think that you should move the skeleton logic here:
https://github.com/safe-global/safe-wallet-monorepo/blob/2fcc6735a94a02cbe227eba5e2c25fb696f6f841/apps/mobile/src/features/Assets/components/Balance/Balance.tsx#L62-L68

Also when you move it there, then I'm not sure we need to display the alert "error while getting the balance of your wallet", since if you have the skeleton it would be clear that it's loading and we keep trying to load the balance every couple of seconds.